app/src/main/java/org/oppia/android/app/devoptions/platformparameters/PlatformParameterItemViewModel.kt

```diff
@@ -0,0 +1,101 @@
+package org.oppia.android.app.devoptions.platformparameters
+
+import androidx.annotation.ColorInt
+import androidx.databinding.ObservableField
+import org.oppia.android.app.model.PlatformParameterId
+import org.oppia.android.app.model.PlatformParameterValue
+import org.oppia.android.app.model.SyncStatus
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.view.models.R
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.locale.OppiaLocale
+
+/**
+ * [ViewModel] for displaying a platform parameter in the RecyclerView of
+ * [PlatformParametersFragment].
+ */
+class PlatformParameterItemViewModel(
+  val platformParameterId: PlatformParameterId,
+  val currentValue: PlatformParameterValue,
+  val syncStatus: SyncStatus,
+  private val machineLocale: OppiaLocale.MachineLocale,
+  private val resourceHandler: AppLanguageResourceHandler
+) : ObservableViewModel() {
+  /** The observable boolean value of the parameter, used for switch toggles in UI. */
+  val isChecked = ObservableField(currentValue.boolean)
+
+  /** Whether the input field should be visible (i.e., if the value is not a boolean). */
+  val isInputVisible = ObservableField(!currentValue.hasBoolean())
+
+  /** The display name of the platform parameter. */
+  val platformParameterDisplayText = ObservableField(getPlatformParameterDisplayName())
```

Comment:
For consistency with variable naming with `getPlatformParmaeterDisplayName()` and across feature flag naming of the same.

```suggestion
  val platformParameterDisplayName = ObservableField(getPlatformParameterDisplayName())
```

Reply:
updated.

Reply:
Could you double-check the naming? 

https://github.com/oppia/oppia-android/blob/908e23a3108fec0aa055b39814a36ae0322d3a66/app/src/main/java/org/oppia/android/app/devoptions/platformparameters/PlatformParameterItemViewModel.kt#L34

The feature flag implementation uses `displayName`, so this differs from that.

https://github.com/oppia/oppia-android/blob/908e23a3108fec0aa055b39814a36ae0322d3a66/app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagItemViewModel.kt#L24-L25

Reply:
I dont know why this error comes when I change the name to `platformParameterDisplayName` but works fine with other names.

`ERROR: Could not find accessor org.oppia.android.app.devoptions.platformparameters.PlatformParameterItemViewModel.platformParameterDisplayName

file:///home/theayushyadav11/.cache/bazel/_bazel_theayushyadav11/42261a1afa518e6c2582297272b65e4d/sandbox/linux-sandbox/3173/execroot/__main__/app/src/main/res/layout/platform_parameter_item.xml Line:34


        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:163)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
        at com.google.devtools.build.buildjar.javac.BlazeJavacMain.compile(BlazeJavacMain.java:137)
        at com.google.devtools.build.buildjar.ReducedClasspathJavaLibraryBuilder.fallback(ReducedClasspathJavaLibraryBuilder.java:105)
        at com.google.devtools.build.buildjar.ReducedClasspathJavaLibraryBuilder.compileSources(ReducedClasspathJavaLibraryBuilder.java:67)
        at com.google.devtools.build.buildjar.SimpleJavaLibraryBuilder.compileJavaLibrary(SimpleJavaLibraryBuilder.java:110)
        at com.google.devtools.build.buildjar.SimpleJavaLibraryBuilder.run(SimpleJavaLibraryBuilder.java:118)
        at com.google.devtools.build.buildjar.BazelJavaBuilder.build(BazelJavaBuilder.java:105)
        at com.google.devtools.build.buildjar.BazelJavaBuilder.parseAndBuild(BazelJavaBuilder.java:85)
        at com.google.devtools.build.lib.worker.WorkRequestHandler$WorkRequestHandlerBuilder.lambda$new$0(WorkRequestHandler.java:287)
        at com.google.devtools.build.lib.worker.WorkRequestHandler$WorkRequestCallback.apply(WorkRequestHandler.java:252)
        at com.google.devtools.build.lib.worker.WorkRequestHandler.respondToRequest(WorkRequestHandler.java:480)
        at com.google.devtools.build.lib.worker.WorkRequestHandler.lambda$startResponseThread$1(WorkRequestHandler.java:433)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: android.databinding.tool.util.LoggedErrorException: Found data binding error(s):

ERROR: Could not find accessor org.oppia.android.app.devoptions.platformparameters.PlatformParameterItemViewModel.platformParameterDisplayName

file:///home/theayushyadav11/.cache/bazel/_bazel_theayushyadav11/42261a1afa518e6c2582297272b65e4d/sandbox/linux-sandbox/3173/execroot/__main__/app/src/main/res/layout/platform_parameter_item.xml Line:34


`

Reply:
Ah, I dont know why Im getting the error when I change the name to `platformParameterDisplayName`, for rest all names everything works fine.

#### Error log:
`ERROR: Could not find accessor org.oppia.android.app.devoptions.platformparameters.PlatformParameterItemViewModel.platformParameterDisplayName

file:///home/theayushyadav11/.cache/bazel/_bazel_theayushyadav11/42261a1afa518e6c2582297272b65e4d/sandbox/linux-sandbox/92/execroot/__main__/app/src/main/res/layout/platform_parameter_item.xml Line:34


        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:163)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
        at com.google.devtools.build.buildjar.javac.BlazeJavacMain.compile(BlazeJavacMain.java:137)
        at com.google.devtools.build.buildjar.ReducedClasspathJavaLibraryBuilder.fallback(ReducedClasspathJavaLibraryBuilder.java:105)
        at com.google.devtools.build.buildjar.ReducedClasspathJavaLibraryBuilder.compileSources(ReducedClasspathJavaLibraryBuilder.java:67)
        at com.google.devtools.build.buildjar.SimpleJavaLibraryBuilder.compileJavaLibrary(SimpleJavaLibraryBuilder.java:110)
        at com.google.devtools.build.buildjar.SimpleJavaLibraryBuilder.run(SimpleJavaLibraryBuilder.java:118)
        at com.google.devtools.build.buildjar.BazelJavaBuilder.build(BazelJavaBuilder.java:105)
        at com.google.devtools.build.buildjar.BazelJavaBuilder.parseAndBuild(BazelJavaBuilder.java:85)
        at com.google.devtools.build.lib.worker.WorkRequestHandler$WorkRequestHandlerBuilder.lambda$new$0(WorkRequestHandler.java:287)
        at com.google.devtools.build.lib.worker.WorkRequestHandler$WorkRequestCallback.apply(WorkRequestHandler.java:252)
        at com.google.devtools.build.lib.worker.WorkRequestHandler.respondToRequest(WorkRequestHandler.java:480)
        at com.google.devtools.build.lib.worker.WorkRequestHandler.lambda$startResponseThread$1(WorkRequestHandler.java:433)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: android.databinding.tool.util.LoggedErrorException: Found data binding error(s):

ERROR: Could not find accessor org.oppia.android.app.devoptions.platformparameters.PlatformParameterItemViewModel.platformParameterDisplayName

file:///home/theayushyadav11/.cache/bazel/_bazel_theayushyadav11/42261a1afa518e6c2582297272b65e4d/sandbox/linux-sandbox/92/execroot/__main__/app/src/main/res/layout/platform_parameter_item.xml Line:34


        at android.databinding.tool.processing.Scope.assertNoError(Scope.java:112)
        at android.databinding.annotationprocessor.ProcessDataBinding.doProcess(ProcessDataBinding.java:124)
        at android.databinding.annotationprocessor.ProcessDataBinding.process(ProcessDataBinding.java:88)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:985)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs(JavacProcessingEnvironment.java:821)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1225)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1362)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1258)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:936)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
        ... 13 more
Target //:oppia_dev failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 55.706s, Critical Path: 54.95s
INFO: 33 processes: 6 internal, 13 linux-sandbox, 14 worker.
FAILED: Build did NOT complete successfully
`

Reply:
This issue arises because data binding looks for a Java-style getter like `getPlatformParameterDisplayName()`. Since there’s already a private method with that exact name, the compiler mistakenly picks it up—but can't access it, leading to an error. To fix this, rename the method (e.g., `retrievePlatformParameterDisplayName`) or pass the ID as a parameter (like in the feature flags screen) to avoid clashing with the getter convention.

Reply:
Ah, thanks for clearing it out, updated now